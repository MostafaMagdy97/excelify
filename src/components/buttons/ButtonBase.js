import styled from 'styled-components';
import { bgColor } from 'styled-system';

import Space from '../../base/Space';

const ButtonBase = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 36px;
  padding: 0 ${Space[8]}px;
  transition: all 200ms ease-out;
  outline: none;
  position: relative;
  min-width: 120px;
  ${bgColor};

  &:not(:disabled) {
    cursor: pointer;
  }
`;

export default ButtonBase;
