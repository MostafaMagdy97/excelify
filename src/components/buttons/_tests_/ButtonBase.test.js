import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import 'jest-styled-components';

import ButtonBase from '../ButtonBase';

configure({ adapter: new Adapter() });

describe('ButtonBase', () => {
  it('should render', () => {
    const component = <ButtonBase>Button</ButtonBase>;
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });
});
