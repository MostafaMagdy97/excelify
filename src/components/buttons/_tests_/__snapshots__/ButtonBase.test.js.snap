// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBase should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 36px;
  padding: 0 24px;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  outline: none;
  position: relative;
  min-width: 120px;
  max-width: -webkit-max-content;
  max-width: -moz-max-content;
  max-width: max-content;
}

.c0:not(:disabled) {
  cursor: pointer;
}

<button
  class="c0"
>
  Button
</button>
`;
